File: Examples/test-suite/perl5/arrays_global_twodim_runme.pl
use strict;
use warnings;
use Test::More tests => 19;
BEGIN { use_ok('arrays_global_twodim') }
require_ok('arrays_global_twodim');

isnt($arrays_global_twodim::array_c, undef, 'array_c');
isnt($arrays_global_twodim::array_sc, undef, 'array_sc');
isnt($arrays_global_twodim::array_uc, undef, 'array_uc');
isnt($arrays_global_twodim::array_s, undef, 'array_s');
isnt($arrays_global_twodim::array_us, undef, 'array_us');
isnt($arrays_global_twodim::array_i, undef, 'array_i');
isnt($arrays_global_twodim::array_l, undef, 'array_l');
isnt($arrays_global_twodim::array_ul, undef, 'array_ul');
isnt($arrays_global_twodim::array_ll, undef, 'array_ll');
isnt($arrays_global_twodim::array_f, undef, 'array_f');
isnt($arrays_global_twodim::array_d, undef, 'array_d');
isnt($arrays_global_twodim::array_struct, undef, 'array_struct');
isnt($arrays_global_twodim::array_structpointers, undef, 'array_structpointers');
isnt($arrays_global_twodim::array_ipointers, undef, 'array_ipointers');
isnt($arrays_global_twodim::array_enum, undef, 'array_enum');
isnt($arrays_global_twodim::array_enumpointers, undef, 'array_enumpointers');
isnt($arrays_global_twodim::array_const_i, undef, 'array_const_i');

File: Examples/test-suite/php/arrays_global_twodim_runme.php
<?php

require "tests.php";

check::functions(array('fn_taking_arrays','get_2d_array',));
check::classes(array('arrays_global_twodim','SimpleStruct','Material'));
check::globals(array('array_c','array_sc','array_uc','array_s','array_us','array_i','array_ui','array_l','array_ul','array_ll','array_f','array_d','array_struct','array_structpointers','array_ipointers','array_enum','array_enumpointers','array_const_i','chitMat','hitMat_val','hitMat'));

$a1=array(10,11,12,13);
$a2=array(14,15,16,17);
$a=array($a1,$a2);

$_a=check::get('array_const_i');

for($x=0;$x<count($a1);$x++) {
  for($y=0;$y<2;$y++) {
    check::equal($a[$y][$x],get_2d_array($_a,$y,$x),"check array $x,$y");
  }
}

check::done();
